### PR TITLE
differentiate the log entries of postscript and postbootscript 

### PR DESCRIPTION
**Pull request by penguhyang**
change log output like this:
postscript
```
Tue Sep 13 07:41:07 EDT 2016 Running PostScript: syslog
Tue Sep 13 07:41:07 EDT 2016 PostScript syslog return with 0
Tue Sep 13 07:41:07 EDT 2016 Running PostScript: remoteshell

Tue Sep 13 07:41:09 EDT 2016 PostScript remoteshell return with 0
```
postbootscript
```
Tue Sep 13 07:41:47 EDT 2016 Running PostBootScript: otherpkgs
./otherpkgs: no extra rpms to install
Tue Sep 13 07:41:47 EDT 2016 PostBootScript otherpkgs return with 0
/xcatpost/mypostscript.post return
```

### CODE REVIEW

File: xCAT-server/share/xcat/install/scripts/post.debian [hunk 1]
```diff
@@ -112,7 +112,7 @@ export PATH=$PATH:/xcatpost
 #  However, Ubuntu cleans /tmp on reboot.  So, for Ubuntu, /var/tmp is used instead.
 
 # use the run_ps subroutine to run the postscripts
-TMP=`sed "/postscripts-start-here/,/postscripts-end-here/ s/\(.*\)/run_ps \1/;s/run_ps\s*#/#/;s/run_ps\s*$//" /xcatpost/mypostscript`
+TMP=`sed "/postscripts-start-here/,/postscripts-end-here/ s/\(.*\)/run_ps \1 PostScript/;s/run_ps\s*#/#/;s/-here\sPostScript/-here/;s/run_ps\s*$//" /xcatpost/mypostscript`
 echo "
 # subroutine used to run postscripts
 run_ps () {
```
Comment (immarvin) [inline xCAT-server/share/xcat/install/scripts/post.debian]: please add some comments on the arguments of ``run_ps``

File: xCAT-server/share/xcat/install/scripts/post.debian [hunk 2]
```diff
@@ -121,19 +121,19 @@ run_ps () {
  logfile=\"/var/log/xcat/xcat.log\"
  
  if [ -f \$1 ]; then 
-  echo "\"\`date\` Running postscript: \$@\"" | tee -a \$logfile
```
Comment (immarvin) [inline xCAT-server/share/xcat/install/scripts/post.debian]: For backward-compatibility, when ``run_ps`` is called with only 1 argument, there should be a default value for $2, such as "postscript"

File: xCAT-server/share/xcat/install/scripts/post.xcat [hunk 3]
```diff
@@ -179,47 +179,44 @@ echo "
 return_value=0
 # subroutine used to run postscripts
 run_ps () {
```
Comment (immarvin) [inline xCAT-server/share/xcat/install/scripts/post.xcat]: please add some comments on the arguments of run_ps

File: xCAT/postscripts/xcatdsklspost [hunk 4]
```diff
@@ -835,8 +835,8 @@ run_ps () {
  local logfile=\"/var/log/xcat/xcat.log\"
 
  if [ -f \$1 ]; then 
-  echo \"\`date\` Running postscript: \$*\" 
-  msgutil_r \"\$MASTER_IP\" \"info\" "\"\`date\` Running postscript: \$*\"" \"\$logfile\"
+  echo \"\`date\` Running PostBootScript: \$*\"
```
Comment (immarvin) [inline xCAT/postscripts/xcatdsklspost]: xcatdsklspost will be called in the following:
````
#   This script is called in the following different places:
#      updatenode -P ... --> xcatdsklspost 1 -m/-M ...
#      updatenode -S --> xcatdsklspost 2 -m/-M otherpkgs
#      moncfg rmcmon --> xcatdsklspost 3 configrmcnodes
#      node deployment  --> xcatdsklspost
#      statelite mode   --> xcatdsklspost 4
#      update security  --> xcatdsklspost 5 -m/-M ...
#      node reboot - xcatpostinit1  --> xcatdsklspost 6
````

for the following 2 modes, the postscripts and post boot scripts should be differentiated
````
#      node deployment  --> xcatdsklspost
#      statelite mode   --> xcatdsklspost 4
````

for the node reboot scenario, should be postbootscript
````
node reboot - xcatpostinit1  --> xcatdsklspost 6
````

for the rest "update node" scenarios, should be "postscript"

File: xCAT-server/share/xcat/install/scripts/post.xcat (hunk 3 above)
Comment (immarvin) [inline xCAT-server/share/xcat/install/scripts/post.xcat]: For backward-compatibility, when run_ps is called with only 1 argument, there should be a default value for $2, such as "postscript"

**Comment by immarvin** [comment]
is it possible to define run_ps in one script and include it in different places?

File: xCAT-server/share/xcat/install/scripts/post.debian (hunk 2 above)
Comment (penguhyang) [inline xCAT-server/share/xcat/install/scripts/post.debian]: fixed.

File: xCAT/postscripts/xcatdsklspost [hunk 5]
```diff
@@ -869,7 +874,11 @@ run_ps () {
 
 " > /$xcatpost/mypostscript
 echo "$TMP" >> /$xcatpost/mypostscript
-TMP=`sed "/# postbootscripts-start-here/,/# postbootscripts-end-here/ s/\(.*\)/run_ps \1/;s/run_ps\s*#/#/;s/run_ps\s*$//" /$xcatpost/mypostscript`
+if [ "$MODE" = "1" ] || [ "$MODE" = "2" ] || [ "$MODE" = "5" ]; then
```
Comment (immarvin) [inline xCAT/postscripts/xcatdsklspost]: what about $mode=3

File: xCAT-server/share/xcat/install/scripts/post.xcat [hunk 6]
```diff
@@ -179,47 +179,48 @@ echo "
...
- # other than root:root. And in some case, the directory /var/log/xcat was
- # created by xCAT, and had root:root ownership. In this way, rsyslogd
- # did not have enough permission to write to log files under this directory.
- # As a dirty hack, change the ownership of directory /var/log/xcat to the
- # same ownership of directory /var/log.
- chown `ls -ld /var/log | awk '{ print \$3\":\"\$4 }'` "\"/var/log/xcat\""
- local logfile=\"/var/log/xcat/xcat.log\"
-
- if [ -f \$1 ]; then
-  echo \"\`date\` Running postscript: \$*\"
```
Comment (immarvin) [inline xCAT-server/share/xcat/install/scripts/post.xcat]: why delete this lines?

File: xCAT-server/share/xcat/install/scripts/post.debian (hunk 1 above)
Comment (penguhyang) [inline xCAT-server/share/xcat/install/scripts/post.debian]: fixed.

File: xCAT-server/share/xcat/install/scripts/post.debian [hunk 7]
```diff
@@ -112,28 +112,32 @@ export PATH=$PATH:/xcatpost
...
+TMP=`sed "/postscripts-start-here/,/postscripts-end-here/ s/\(.*\)/run_ps \1 postscript/;s/run_ps\s*#/#/;s/-here\spostscript/-here/;s/run_ps\s*$//" /xcatpost/mypostscript`
 echo "
 # subroutine used to run postscripts
 run_ps () {
  logdir=\"/var/log/xcat\"
  mkdir -p \$logdir
  logfile=\"/var/log/xcat/xcat.log\"
+
+ if [ ! \$2 ]; then
+  \$2=\"postscript\"
```
Comment (penguhyang) [inline xCAT-server/share/xcat/install/scripts/post.debian]: fixed.

File: xCAT/postscripts/xcatdsklspost [hunk 8]
```diff
@@ -834,9 +838,12 @@ run_ps () {
  mkdir -p "\"/var/log/xcat\""
  local logfile=\"/var/log/xcat/xcat.log\"
 
+ if [ ! \$2 ]; then
```
Comment (penguhyang) [inline xCAT/postscripts/xcatdsklspost]: fixed.

File: xCAT-server/share/xcat/install/scripts/post.xcat (hunk 6 above)
Comment (penguhyang) [inline xCAT-server/share/xcat/install/scripts/post.xcat]: fixed.

File: xCAT-server/share/xcat/install/scripts/post.sles11.s390x [hunk 9]
```diff
@@ -101,28 +101,34 @@ PATH=$PATH:/xcatpost
...
 echo "
 # subroutine used to run postscripts
+# $1 argument is the script name
+# $2 argument is the script type
 run_ps () {
  logdir=\"/var/log/xcat\"
  mkdir -p \$logdir
  logfile=\"/var/log/xcat/xcat.log\"
+
+ if [ ! \$2 ]; then
```
Comment (immarvin) [inline xCAT-server/share/xcat/install/scripts/post.sles11.s390x]: should be be``if [ -z "$2"]``?

File: xCAT/postscripts/xcatdsklspost [hunk 10]
```diff
@@ -869,7 +879,11 @@ run_ps () {
 
 " > /$xcatpost/mypostscript
 echo "$TMP" >> /$xcatpost/mypostscript
-TMP=`sed "/# postbootscripts-start-here/,/# postbootscripts-end-here/ s/\(.*\)/run_ps \1/;s/run_ps\s*#/#/;s/run_ps\s*$//" /$xcatpost/mypostscript`
+if [ "$MODE" = "1" ] || [ "$MODE" = "2" ]  || [ "$MODE" = "3" ] || [ "$MODE" = "5" ]; then
+    TMP=`sed "/# postbootscripts-start-here/,/# postbootscripts-end-here/ s/\(.*\)/run_ps \1 postscript/;s/run_ps\s*#/#/;s/-here\spostscript/-here/;s/run_ps\s*$//" /$xcatpost/mypostscript`
```
Comment (immarvin) [inline xCAT/postscripts/xcatdsklspost]: what's ``s/run_ps\s*#/#/;s/-here\spostscript/-here/`` for?

**Comment by immarvin** [comment]
agree to merge

**Pull request merged**